### ISSUE TITLE
0.6.0 does not log in when a user does not have any programmable gates

### ISSUE THREAD

**Issue by avano** [Pigotka/ha-cc-jablotron-cloud#20]
when trying to log in with the 0.6.0 version I only get:

```
2024-06-17 12:12:37.407 DEBUG (MainThread) [custom_components.jablotron_cloud] Preparing Jablotron data update coordinator
2024-06-17 12:12:39.433 DEBUG (MainThread) [custom_components.jablotron_cloud] Failed to get gates data for service <service id>
2024-06-17 12:12:39.433 DEBUG (MainThread) [custom_components.jablotron_cloud] Finished fetching Jablotron Cloud data in 2.025 seconds (success: False)
```

It fails on [this line](https://github.com/Pigotka/ha-cc-jablotron-cloud/blob/main/custom_components/jablotron_cloud/__init__.py#L141)

calling the `get_programmable_gates` manually using jablotronpy library, it returns `{'service-states': {'last-event-time': '2024-06-17T12:08:29+0200', 'service-name': '<name>'}, 'states': []}` (with adding `print(data)` there) and raises the exception [here](https://github.com/fdegier/JablotronPy/blob/main/jablotronpy/jablotronpy.py#L280)

In the jablotron UI I only have 1 `section` and 1 `keypad` - previously I used the `section` to arm/disarm the alarm using this integration

**Comment by Pigotka** [comment]
> The design principle was to either return data or an UnexpectedResponse as we are returning the entire data from Jablotron instead of parsing the PG's

> We could change it so it returns only a list of programmableGates or an empty list. But IIRC you also needed the states in order to map the cloud component ID?

The problem is that I'm not able to recognize connection fail from not having the gates. For thermo devices it works well so I would find it resonable to have it unified.

**Comment by fdegier** [comment]
> > The design principle was to either return data or an UnexpectedResponse as we are returning the entire data from Jablotron instead of parsing the PG's
> 
> > We could change it so it returns only a list of programmableGates or an empty list. But IIRC you also needed the states in order to map the cloud component ID?
> 
> @fdegier The problem is that I'm not able to recognize connection fail from not having the gates. For thermo devices it works well so I would find it resonable to have it unified.
> 
> I suggest to return the data whenever they are valid. For gates it would mean for example that 'states' is present in the data.

For thermo devices we return `states` and not a key specific to thermo devices. And that could be an empty list.

**Comment by Pigotka** [comment]
Right, my interation does not need full data and only proramableates would be enough but I do not want you to do such breakin chanes. 
I still think you can easily detect whether the response was valid by checking `states` key instead of `programmableGates` to allow returnin the data with empty programmableGates.

**Comment by fdegier** [comment]
> Right, my interation does not need full data and only programmableGates would be enough but I do not want you to do such breakin changes. I still think you can easily detect whether the response was valid by checking `states` key instead of `programmableGates` to allow returnin the data with empty programmableGates.

The thing is, if a user doesn't have programmable keys, the key is not even present. So I would return a dict with only `states`.

**Comment by fdegier** [comment]
> Exactly. As you mentioned earlier. For gates you return all data so this is correct. I'm already checking presence of `programmableGates` so it would work just fine.

That is the current situation: https://github.com/fdegier/JablotronPy/blob/255fd4d7fbadbf46da4dc919394cbb393330dc9d/jablotronpy/jablotronpy.py#L278

**Comment by Pigotka** [comment]
> > Exactly. As you mentioned earlier. For gates you return all data so this is correct. I'm already checking presence of `programmableGates` so it would work just fine.
> 
> That is the current situation: https://github.com/fdegier/JablotronPy/blob/255fd4d7fbadbf46da4dc919394cbb393330dc9d/jablotronpy/jablotronpy.py#L278

And I suggest:
`if status and 'states' in data:`

**Comment by Pigotka** [comment]
I'm trying not to introduce breaking changes so I suggest to keep the data format and fix the code so it only throws when the API does not respond properly. It means checkin status and if necessary also presence of some key in data but for sessions, sections and gates it might be optional as the existence of the key is not required there.

**Comment by fdegier** [comment]
New version 0.6.3 allows empty PG

**Comment by Pigotka** [comment]
Included in new release.

**Comment by avano** [comment]
Just for the record, the bump of `jablotronpy` library is not included in `0.6.1` release, but after changing it manually in my installation it works OK
